lib/3scale/backend/stats/cleaner.rb

```diff
@@ -245,6 +252,16 @@ def delete_stats_keys_with_val_0(redis_conn, log_deleted_keys)
               sleep(SLEEP_BETWEEN_SCANS)
             end
           end
+
+          def with_retries
+            begin
+              retries ||= 0
+              return yield
+            rescue Exception => e
+              retry if (retries += 1) < MAX_RETRIES_REDIS_ERRORS
```

Comment:
:astonished: This line is a bit surprising. How is this following your usual readability policy? Why don't you just extract the increment to a statement before the `retry` check?

Reply:
I don't have a strong opinion about this. You would have complained either way :trollface: